src/main/java/com/commercetools/sync/shoppinglists/ShoppingListSync.java

```diff
@@ -0,0 +1,300 @@
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * This class syncs shopping list drafts with corresponding shopping list resources in CTP project.
+ */
+public class ShoppingListSync extends BaseSync<ShoppingListDraft, ShoppingListSyncStatistics, ShoppingListSyncOptions> {
+
+    private static final String CTP_SHOPPING_LIST_UPDATE_FAILED =
+        "Failed to update shopping lists with key: '%s'. Reason: %s";
+    private static final String CTP_SHOPPING_LIST_FETCH_FAILED = "Failed to fetch existing shopping lists with keys: "
+        + "'%s'.";
+    private static final String FAILED_TO_PROCESS = "Failed to process the ShoppingListDraft with key:'%s'. Reason: %s";
+
+
+    private final ShoppingListService shoppingListService;
+    private final ShoppingListReferenceResolver shoppingListReferenceResolver;
+    private final ShoppingListBatchValidator shoppingListBatchValidator;
+
+    /**
+     * Takes a {@link ShoppingListSyncOptions} to instantiate a new {@link ShoppingListSync} instance that could be used
+     * to sync shopping list drafts in the CTP project specified in the injected {@link ShoppingListSyncOptions}
+     * instance.
+     *
+     * @param shoppingListSyncOptions the container of all the options of the sync process including the CTP project
+     *                                client and/or configuration and other sync-specific options.
+     */
+    public ShoppingListSync(@Nonnull final ShoppingListSyncOptions shoppingListSyncOptions) {
+
+        this(shoppingListSyncOptions,
+            new ShoppingListServiceImpl(shoppingListSyncOptions),
+            new CustomerServiceImpl(shoppingListSyncOptions),
+            new TypeServiceImpl(shoppingListSyncOptions));
+    }
+
+    /**
+     * Takes a {@link ShoppingListSyncOptions} and service instances to instantiate a new {@link ShoppingListSync}
+     * instance that could be used to sync shopping list drafts in the CTP project specified in the injected
+     * {@link ShoppingListSyncOptions} instance.
+     *
+     * <p>NOTE: This constructor is mainly to be used for tests where the services can be mocked and passed to.
+     *
+     * @param syncOptions   the container of all the options of the sync process including the CTP project
+     *                      client and/or configuration and other sync-specific options.
+     * @param shoppingListService   the shopping list service which is responsible for fetching/caching the
+     *                              ShoppingLists from the CTP project.
+     * @param customerService   the customer service which is responsible for fetching/caching the Customers from the
+     *                          CTP project.
+     * @param typeService   the type service which is responsible for fetching/caching the Types from the CTP project.
+     */
+    protected ShoppingListSync(@Nonnull final ShoppingListSyncOptions syncOptions,
```

Comment:
we can make it as package-default access modifier. 
```suggestion
   ShoppingListSync(@Nonnull final ShoppingListSyncOptions syncOptions,
```

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/615/commits/939a5993f882d9d508c16ce80c272603db27d062